pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -396,6 +442,11 @@ func (ctrl *PersistentVolumeController) syncUnboundClaim(claim *v1.PersistentVol
 		if err != nil {
 			return err
 		}
+		// Clean up the waiting for consumer condition as we already have VolumeName set
+		// If the condition was found and cleared, reconcile one more time to refresh the cache.
+		if foundCondition, err := ctrl.cleanWaitingForConsumerCondition(claim); foundCondition {
+			return err
+		}
```

Comment:
is this the only code path that should clean up this condition? there's a lot of branches in syncUnboundClaim (in addition to syncBoundClaim and updateClaimMigrationAnnotations), and it's not clear to me that all of them should leave a WaitingForConsumer=true condition present on the claim